src/test/java/com/sproutsocial/nsq/BaseDockerTestIT.java

```diff
@@ -0,0 +1,131 @@
+
+        scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    @After
+    public void teardown() throws InterruptedException {
+        client.stop();
+
+        cluster.shutdown();
+
+        scheduledExecutorService.shutdown();
+        scheduledExecutorService.awaitTermination(10, TimeUnit.SECONDS);
+    }
+
+    protected void send(String topic, List<String> msgs, double delayChance, int maxDelay, Publisher publisher) {
+        int count = 0;
+        LOGGER.info("Sending {} messags to topic {}", msgs.size(), topic);
+        Random random = new Random();
+        for (String msg : msgs) {
+            if (random.nextFloat() < delayChance) {
+                Util.sleepQuietly(random.nextInt(maxDelay));
+            }
+            publisher.publish(topic, msg.getBytes());
+            if (++count % 10 == 0) {
+                System.out.println("sent " + count + " msgs");
+            }
+        }
+    }
+
+    protected List<String> messages(int count, int len) {
+        List<String> res = new ArrayList<>(count);
+        int batch = messageBatchCounter.getAndIncrement();
+        Random random = new Random();
+        for (int i = 0; i < count; i++) {
+            StringBuilder s = new StringBuilder();
+            s.append(String.format("msg %04d batch: %04d len:%04d ", i, batch, len));
+            for (int j = 0; j < len; j++) {
+                s.append((char) (33 + random.nextInt(92)));
+            }
+            res.add(s.toString());
+        }
+        return res;
+    }
+
+    protected void sendAndVerifyMessagesFromBackup(Publisher publisher, TestMessageHandler handler) {
+        List<String> postFailureMessages = messages(20, 40);
+        send(topic, postFailureMessages, 0.5, 10, publisher);
+        List<NSQMessage> receivedMessages = handler.drainMessagesOrTimeOut(postFailureMessages.size());
+        validateReceivedAllMessages(postFailureMessages, receivedMessages, true);
+        validateFromParticularNsqd(receivedMessages, 1);
+    }
+
+    protected void sendAndVerifyMessagesFromPrimary(Publisher publisher, TestMessageHandler handler) {
+        List<String> messages = messages(20, 40);
+        send(topic, messages, 0.5f, 10, publisher);
+        List<NSQMessage> receivedMessages = handler.drainMessagesOrTimeOut(messages.size());
+        validateReceivedAllMessages(messages, receivedMessages, true);
+        validateFromParticularNsqd(receivedMessages, 0);
+    }
+
+    protected void validateFromParticularNsqd(List<NSQMessage> receivedMessages, int nsqHostIndex) {
+        for (NSQMessage e : receivedMessages) {
+            Assert.assertEquals(cluster.getNsqdNodes().get(nsqHostIndex).getTcpHostAndPort(), e.getConnection().getHost());
+        }
+    }
+
+    protected Publisher primaryOnlyPublisher() {
+        return new Publisher(client, cluster.getNsqdNodes().get(0).getTcpHostAndPort().toString(), null);
+    }
+
+    protected Publisher backupPublisher() {
```

Comment:
NIT: This is not a backup publisher but a publisher with backup NSQD, so may be `publisherWithBackup` as a name?